sched/scheduler/jobRunner.go

```diff
@@ -0,0 +1,267 @@
+package scheduler
+
+import (
+	"fmt"
+	"github.com/scootdev/scoot/saga"
+	"github.com/scootdev/scoot/sched"
+	cm "github.com/scootdev/scoot/sched/clustermembership"
+	dist "github.com/scootdev/scoot/sched/distributor"
+	"sync"
+)
+
+const MAX_RETRY = 3
+
+type jobRunner struct {
+	job       sched.Job
+	saga      saga.Saga
+	sagaState *saga.SagaState
+	dist      *dist.PoolDistributor
+
+	updateChan chan stateUpdate
+	wg         sync.WaitGroup
+}
+
+func NewJobRunner(job sched.Job, saga saga.Saga, sagaState *saga.SagaState, nodes []cm.Node) jobRunner {
+	jr := jobRunner{
+		job:        job,
+		saga:       saga,
+		sagaState:  sagaState,
+		dist:       dist.NewPoolDistributor(cm.StaticClusterFactory(nodes)),
+		updateChan: make(chan stateUpdate),
+	}
+
+	go jr.updateSagaState()
+	return jr
+}
+
+// Runs the Job associated with this JobRunner to completion
+func (jr jobRunner) runJob() {
```

Comment:
This is great to get going.

This is the loop that I've been saying could be decomposed to be an explicit function at each tick. (It's easier to debug and test a function with explicit input and output than a long-running function that accretes state and writes updates).

Would it make sense for me to try that reorientation after you merge this, and you can see if you think it makes more sense that way?